### ISSUE TITLE
[Enhancement] – In the TOC – Table of contents menu, please had the normal keys page up and page down.

### ISSUE THREAD

**Issue by joaocarvalhoopen** [ahrm/sioyek#191]
Hello Ahrm,

I experimented opening the ARRL Handbook 2022 it’s a big PDF book 1532 pages with many images and diagram. It opens well and works well. It uses 1.0 GB of RAM, but it’s a big book.

This book has a big index and what I have found is that it’s a bit of a pain to make arrow down to see the index so it would be really nice, if the **keys Page Down** and **Page Up** worked **inside the TOC - Table Of Contents**  menu.

Note: I know that you can filter terms, but Page Up and Page down in this context make a lot of sense.

The very best regards,
João

**Comment by ahrm** [comment]
Added in 85772bae0087a297032af90c269896bf191ee5c7.

**Comment by joaocarvalhoopen** [comment]
**Thank very much you Ahrm**, when the development build comes out I will test it in Linux Ubuntu 20.04. 

This feature is really useful.

Yesterday I installed Sioyek on other family member computer in windows and everything it worked really well. Friday I will install it on even other family member in Linux.
She also liked it very much, especially the custom colors of my settings for custom_color dyslexic and the one line attention mode. Things like the Drag and Drop and the mouse point zoom where also things that she liked. I showed her the F11 – Full screen mode to read in a very clean way long books, and that also was a winner. This member of my family that is studying architecture in college, so she has to read also a lot.

Note: I’m installing every one the development builds, because I consider the last features that you have implemented really important! And because from my experience at least your last builds have been really stable.

Thank you.

The very best regards,
João

**Comment by joaocarvalhoopen** [comment]
Hello Ahrm,

I tested the new Page up and page down features in Linux Ubuntu 20.04, and they worked, then I configured the prefs_user with my copy do my configs from the site and I configure the keys_user with  my configs from the site and I, open up the ARRL Handbook 2022 and enter the TOC and all hell break lose soddenly I had one core at 100 % and the App working, I closed it and the process was still running buit the window closed. Then made top and kill -9 -1 and my Ubuntu session closed, made a re initialization to the computer and things get to normal with my computer, but I **think you have introduced a bug in this version.**

The page up and page down worked well in TOC and inside the Ctrl + O file system.


Thank you,

The very best regards,
João

**Comment by joaocarvalhoopen** [comment]
I was feeling brave, so I tested it again and I think that what you have in the code is a race condition between the loading of a really big PDF that takes seconds to pre-render it and the immediate pressing the key “T” to enter TOC and navigating inside the TOC and sending it to a particular page inside the TOC, before it had time to load / pre-render all the PDF. 

This is what I can help you with,

Thank you,

The very best regards,
João

**Comment by ahrm** [comment]
Thanks for your investigation. I don't think this bug was introduced in the last commits though, it probably always existed.
I failed to reproduce the issue on a 1200-page 900mb PDF file. Is the PDF file you are having problem with openly available on the web so I can test it?

**Comment by joaocarvalhoopen** [comment]
Unfortunately, the book download comes only when you buy the paper book and it’s not openly accessible. Sorry.

This book is not only big 1532 pag but also has many images and many diagrams, so it is processing in the background after you open it for some 15 to 20 seconds with the processor at 100 %, but then the CU processing comes to 0 %. And it was while this book was processing at 100 % after loading that I used the TOC and the above bug appened. So I attributed it to some kind of race condition. 

But if I seen the some problem with a open accessible PDF I tell you.

Thank you,

The very best regards,
João
